### ISSUE TITLE
Assigning workspace to monitors does throws i3 config error

### ISSUE THREAD

**Issue by SolidEva** [regolith-linux/regolith-i3#12]
If I add the following to my config-4.16-1ubuntu18ppa10 to assign workspaces to specific monitors, which works fine in normal i3:

```
# Assign workspaces to monitors. Uses the colomns on a numpad for assignment to each monitor
# Check these labels with xrandr
set $monitor_one "DP-1-1-2"
set $monitor_two "DP-1-1-1"
workspace $ws1 output $monitor_one
workspace $ws4 output $monitor_one
workspace $ws7 output $monitor_one
workspace $ws2 output $monitor_two
workspace $ws5 output $monitor_two
workspace $ws8 output $monitor_two
workspace $ws3 output $monitor_two
workspace $ws6 output $monitor_two
workspace $ws9 output $monitor_two
#workspace $ws0 aka 10 is a free floater

```

The following errors are thrown by i3:

```
ERROR: CONFIG: Expected one of these tokens: 'output', 'gaps'
ERROR: CONFIG: (in file /home/emcginnis/.config/i3-regolith/config-4.16-1ubuntu18ppa10)
ERROR: CONFIG: Line  35: set "DP-1-1-2" "DP-1-1-2"
ERROR: CONFIG: Line  36: set "DP-1-1-1" "DP-1-1-1"
ERROR: CONFIG: Line  37: workspace 1:<span font_desc="Source Code Pro Medium 13"> 1 </span><span foreground="#268bd2"></span> output "DP-1-1-2"
ERROR: CONFIG:                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: CONFIG: Line  38: workspace 4:<span font_desc="Source Code Pro Medium 13"> 4 </span><span foreground="#b58900"></span> output "DP-1-1-2"
ERROR: CONFIG: Line  39: workspace 7:<span font_desc="Source Code Pro Medium 13"> 7 </span><span foreground="#d33682"></span> output "DP-1-1-2"
ERROR: CONFIG: Expected one of these tokens: 'output', 'gaps'
ERROR: CONFIG: (in file /home/emcginnis/.config/i3-regolith/config-4.16-1ubuntu18ppa10)
ERROR: CONFIG: Line  36: set "DP-1-1-1" "DP-1-1-1"
ERROR: CONFIG: Line  37: workspace 1:<span font_desc="Source Code Pro Medium 13"> 1 </span><span foreground="#268bd2"></span> output "DP-1-1-2"
ERROR: CONFIG: Line  38: workspace 4:<span font_desc="Source Code Pro Medium 13"> 4 </span><span foreground="#b58900"></span> output "DP-1-1-2"
```
if I replace  `$ws#` with `i3-wm.workspace.0#.name` the errors go away

**Comment by SolidEva** [comment]
Update: while replacing `$ws#` with `i3-wm.workspace.0#.name` makes the errors go away, it does not fix the issue as workspaces still don't get assigned to the proper monitors. Removing the `.name` part does nothing to improve this either. 

I'm guessing that i3 doesn't escape part of the font and Xresource information added in the workspace variables.

**Comment by kgilmer** [comment]
I agree with your assessment @SolidHal .  The Xresource substitutions can be finicky due to the nature of the C preprocessor.  Let me play around with the config on my machine and see if I can get it working.

**Comment by kgilmer** [comment]
None of the obvious quoting tricks seem to help i3 parse the Pango markup in the workspace name.  Looking at the i3 user guide, it seems that maybe we can reference the index of the workspace rather than the name (which is really a label):
```
If you use named workspaces, they must be quoted:

Examples:

workspace 1 output LVDS1
...
```

The problem is that the quote characters within the name of the workspace are not escaped before processing the whole command, I'm guessing.

Does it work if you just use index numbers instead of the workspace names, as quoted above?

I am on the road and don't have access to multiple monitors to test, but at least it doesn't fail to parse :).

**Comment by p3k** [comment]
fyi: referring to the workspaces with index numbers solves the issue for me – https://github.com/regolith-linux/regolith-desktop/issues/84#issuecomment-520169715

**Comment by SolidEva** [comment]
Using the numbers works fine as long as you switch to the "proper" workspaces using the keycommands before opening any windows when you have a new, empty i3 session (new login). 

If you don't, you'll be in workspace 1, but not workspace 1 [colored square]. same goes for 2,3,etc on any additional monitors.

Not sure if this is related to this issue, or if it is a separate bug.

**Comment by majorhayes** [comment]
I'm having an issue with this as well.  Using just numbers instead of `$ws1` also results in separate workspaces named `1` and `1 [ ]`.

**Comment by jtcrank** [comment]
I too am having issues with this. Has anyone found a proper solution to this.

**Comment by vlahan** [comment]
Has there been any progress on resolving this or finding a usable workaround? Not being able to pre-assign workspaces to specific displays significantly reduces the usability of regolith

**Comment by kgilmer** [comment]
Hello @vlahan :smile: , I just tried this with i3-gaps-wm 4.17.1 and the feature works as expected.  Here is my configuration:

```
$ i3 --moreversion
Binary i3 version:  4.17.1 (2019-08-30) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.17.1 (2019-08-30) (pid 5607)abort…)
Loaded i3 config: /home/kgilmer/.config/regolith/i3/config (Last modified: Mon 25 Nov 2019 09:35:46 PM PST, 39061 seconds ago)
```

```
set $monitor_one "DP-0"
set $monitor_two "DP-2"
workspace 1 output $monitor_one
workspace 4 output $monitor_one
workspace 7 output $monitor_one
workspace 2 output $monitor_two
workspace 5 output $monitor_two
workspace 8 output $monitor_two
workspace 3 output $monitor_two
workspace 6 output $monitor_two
workspace 9 output $monitor_two
```

I'm guessing that this is a fix between i3-gaps-wm 4.16 which is in PPA `regolith-stable` and 4.17.1 which is in `regolith-unstable` but have not verified the fix doesn't work on 4.16.  If you would like to get i3-gaps-wm 4.17.1 before the Regolith 1.3 release is out, you can install it via the [speed-ricer](https://github.com/regolith-linux/speed-ricer) [PPA](https://launchpad.net/~kgilmer/+archive/ubuntu/speed-ricer).

If you find this issue is not resolved for you after upgrading i3-gaps-wm kindly reopen this ticket.

**Comment by vlahan** [comment]
Hi Ken,

I upgraded to 4.17.1:
```
$ i3 --moreversion
Binary i3 version:  4.17.1 (2019-08-30) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.17.1 (2019-08-30) (pid 5811)abort…)
Loaded i3 config: /home/vlahan/.config/regolith/i3/config 
```
The initial placement seems to be working well now. However, there is still a glitch similar to the issues described in https://github.com/regolith-linux/regolith-i3/issues/12#issuecomment-520561073 and in https://github.com/regolith-linux/regolith-i3/issues/12#issuecomment-530088444

For example, I have the following rules in my i3 config:

```
set $m_laptop "eDP-1"
set $m_present "HDMI-1"
...
workspace 7 output $m_present $m_laptop
...
for_window [title="pdfpc - presentation.*"] move container to workspace 7
```
When I start pdfpc, the above rule ends up creating what appears to be a duplicate workspace 7:
 
![Screenshot from 2019-11-27 11-24-00](https://user-images.githubusercontent.com/2195338/69715874-49dfc100-1109-11ea-9882-21c83570a23c.png)
 workspace 7.

However, when I move to workspace 7, this phantom duplicate seems to disappear.  

Don't get confused by the way the workspace name looks, I have removed the color from my Xresources definition, only the font span remains):

```
#define WORKSPACE_NAME(INDEX, FONT) INDEX:<span font_desc=FONT>INDEX</span>
...
i3-wm.workspace.07.name: WORKSPACE_NAME(7, glyph_font)
```
